app/controllers/provider_interface/decisions_controller.rb

```diff
@@ -125,5 +155,36 @@ def defer_offer
     def make_an_offer_params
       params.require(:make_an_offer)
     end
+
+    def offer_context_params(course_option)
+      {
+        course_id: course_option.course.id,
+        course_option_id: course_option.id,
+        provider_id: course_option.provider.id,
+        study_mode: course_option.study_mode,
+        location_id: course_option.site.id,
+        decision: :default,
+        standard_conditions: MakeAnOffer::STANDARD_CONDITIONS,
+      }
+    end
+
+    def application_choice_allowed_to_make_decision
```

Comment:
This method could be more descriptively named as `redirect_back_if_application_in_decision_pending_state`

Reply:
Having said that, this check looks like it would be the wrong way around. Also, the `status` is a string where the array contains symbols only.

Can we add a spec to cover this behaviour?

Reply:
https://github.com/DFE-Digital/apply-for-teacher-training/pull/4218/files#diff-437c905cf8db3c525696f24fdd6c586ba729cbe555aa9eeb6a0838484478726b